core/controllers/suggestion.py

```diff
@@ -68,14 +68,19 @@ def put(self, exploration_id, suggestion_id):
             raise self.InvalidInputException('This handler allows actions only'
                                              ' on suggestions to explorations.')
 
-        if suggestion_id.split('.')[1] != exploration_id:
+        if suggestion_id.split('.')[1] != target_id:
             raise self.InvalidInputException('The exploration id provided does '
                                              'not match the exploration id '
                                              'present as part of the '
                                              'suggestion_id')
 
         action = self.payload.get('action')
         suggestion = suggestion_services.get_suggestion_by_id(suggestion_id)
+
+        if suggestion.author_id == self.user_id:
+            raise self.UnauthorizedUserException('You cannot perform an action '
```

Comment:
"perform an action" is quite vague. Maybe "accept/reject"?

Reply:
Done.